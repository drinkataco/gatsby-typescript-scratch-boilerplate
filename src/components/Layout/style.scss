main {
  @apply py-3;

  h1 {
    @apply text-2xl mb-2;
  }

  h2 {
    @apply text-xl;
  }

  h3 {
    @apply text-lg;
  }

  p {
    @apply my-3;
  }

  a {
    @apply text-blue-400 underline;
  }

  ol {
    @apply list-decimal list-inside ml-4;
  }

  ul {
    @apply list-disc list-inside ml-4;
  }

  blockquote {
    @apply border-l-2 pl-4 text-slate-600 border-slate-400 my-5 block;
  }

  iframe {
    @apply my-3;
  }
}
