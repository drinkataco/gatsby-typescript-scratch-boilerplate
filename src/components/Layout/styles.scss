@import '../../assets/scss/variables';

.layout {
  width: $layout-width;
  margin: 0 auto;
}
