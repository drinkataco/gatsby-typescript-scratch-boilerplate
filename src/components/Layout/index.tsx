import * as React from 'react';
import Helmet from 'react-helmet';

import Header from '../Header';
import { SiteMetadata, useSiteMetadata } from '../../hooks/useSiteMetadata';

import './style.scss';

interface LayoutProps {
  children: React.ReactNode | React.ReactNode[];
  title?: string;
  description?: string;
}

function MainLayout({ children, description, title }: LayoutProps) {
  const siteTitle = 'Gatsby Starter Template';
  const meta: SiteMetadata = useSiteMetadata();
  const metaDescription = description || meta.description;

  return (
    <>
      <Helmet>
        <html lang="en" />
        <title>{siteTitle && (title || meta.title)}</title>
        {metaDescription && (
          <meta name="description" content={metaDescription} />
        )}
      </Helmet>
      <div className="w-4/5 m-auto font-sans pb-3">
        <Header siteTitle={ siteTitle } />
        <main>
          {children}
        </main>
      </div>
    </>
  );
}

export default MainLayout;
