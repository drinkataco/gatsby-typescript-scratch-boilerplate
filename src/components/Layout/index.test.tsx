import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';

import Layout from './index';

describe('Layout', () => {
  const props = {
    children: 'test',
    description: 'test',
    title: 'test',
  };

  beforeEach(() => {
    (StaticQuery as jest.Mock).mockImplementationOnce(
      ({ render }) => render(props),
    );
    (useStaticQuery as jest.Mock).mockReturnValue({ site: { siteMetadata: {} } });
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Layout>
          <h1>Test</h1>
        </Layout>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
