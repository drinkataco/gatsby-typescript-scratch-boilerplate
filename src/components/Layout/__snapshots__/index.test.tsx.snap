// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="layout"
>
  <h1>
    Test
  </h1>
</div>
`;

exports[`Layout renders correctly 1`] = `
<div
  className="layout"
>
  <h1>
    Test
  </h1>
</div>
`;
